app/models/form.rb

```diff
@@ -14,12 +14,19 @@ class Form < ActiveRecord::Base
   has_paper_trail on: [:update, :destroy]
   has_many :form_elements, -> { order(:position) }
 
+  after_touch do
+    [Plugins::Petition, Plugins::Fundraiser].each do |plugin_type|
+      plugin_type.where(form_id: id).each do |plugin|
+        plugin.page.touch
+      end
+    end
+  end
```

Comment:
I would lean toward the polymorphic `belongs_to`. It's definitely the most standard Rails way to do this association.